client/app/data/dashboard/bceid/select_operator.json

```diff
@@ -3,7 +3,11 @@
     "title": "Select Operator",
     "content": "View or update your operator information",
     "links": [
-      { "title": "Select Operator ➤", "href": "/dashboard/select-operator" }
+      {
+        "title": "Select Operator ➤",
+        "href": "/dashboard/select-operator",
+        "notification": "operator"
+      }
```

Comment:
For you consideration, instead of adding a `notification` property; perhaps, you can leverage the `href` property and change the `link.title` in the notification function...
Something like:
```
"use client";

import {
  FrontEndRoles,
  OperatorStatus,
  UserOperatorStatus,
} from "@/app/utils/enums";
import reportAProblemTile from "@/app/data/dashboard/report_a_problem.json";
import bceidSelectOperatorTile from "@/app/data/dashboard/bceid/select_operator.json";
import bceidMyOperatorTile from "@/app/data/dashboard/bceid/my_operator.json";
import bceidOperationsTile from "@/app/data/dashboard/bceid/operations.json";
import bceidUsersTile from "@/app/data/dashboard/bceid/users.json";
import idirOperatorsTile from "@/app/data/dashboard/idir/operators.json";
import idirOperationsTile from "@/app/data/dashboard/idir/operations.json";
import idirUsersTile from "@/app/data/dashboard/idir/users.json";
import Inbox from "@/app/components/icons/Inbox";
import Layers from "@/app/components/icons/Layers";
import Notification from "@/app/components/icons/Notification";
import Users from "@/app/components/icons/Users";
import Wrench from "@/app/components/icons/Wrench";

// 📐 type for ContentItem used to build dashboard content tiles
type ContentItem = {
  title: string;
  content: string;
  links?: { title: string; href: string; notification?: string }[];
};

const iconsMap: Record<string, any> = {
  "My Operator": Inbox,
  "My Operations": Layers,
  Operators: Inbox,
  "Select Operator": Inbox,
  Operations: Layers,
  "Report a Problem": Wrench,
  Users: Users,
  "User Access Management": Users,
};

const userOperatorNotifications = (operatorStatus: string | undefined) => {
  let notifications = 0;
  if (!operatorStatus || operatorStatus === OperatorStatus.DRAFT) {
    notifications++;
  }
  return notifications.toString() + " pending action(s) required";
};

// notificationMap is used to retrieve the notification count for each type of notification
// set the notification type in the links array of the Tile JSON file
const notificationMap: Record<string, Function> = {
  "/dashboard/select-operator": userOperatorNotifications,
  // Add Operations and User notification functions here
};
export default function Tiles({
  role,
  operatorStatus,
  userOperatorStatus,
}: {
  role: string;
  operatorStatus: string;
  userOperatorStatus: string;
}) {
  let contents: ContentItem[] | null = null;
  switch (role) {
    case FrontEndRoles.CAS_ADMIN:
      contents = [
        ...idirOperatorsTile,
        ...idirOperationsTile,
        ...idirUsersTile,
        ...reportAProblemTile,
      ];

      break;
    case FrontEndRoles.CAS_ANALYST:
      contents = [
        ...idirOperatorsTile,
        ...idirOperationsTile,
        ...reportAProblemTile,
      ];
      break;
    case FrontEndRoles.INDUSTRY_USER_ADMIN:
      if (userOperatorStatus === UserOperatorStatus.APPROVED) {
        if (
          operatorStatus === OperatorStatus.PENDING ||
          operatorStatus === OperatorStatus.APPROVED
        ) {
          contents = [
            ...bceidMyOperatorTile,
            ...bceidOperationsTile,
            ...bceidUsersTile,
            ...reportAProblemTile,
          ];
        } else {
          contents = [...bceidSelectOperatorTile, ...reportAProblemTile];
        }
        break;
      }
    case FrontEndRoles.INDUSTRY_USER:
      if (userOperatorStatus === UserOperatorStatus.APPROVED) {
        if (
          operatorStatus === OperatorStatus.PENDING ||
          operatorStatus === OperatorStatus.APPROVED
        ) {
          contents = [
            ...bceidMyOperatorTile,
            ...bceidOperationsTile,
            ...reportAProblemTile,
          ];
        }
      } else {
        contents = [...bceidSelectOperatorTile, ...reportAProblemTile];
      }
      break;
  }

  return (
    <section className="flex flex-wrap gap-x-16 lg:gap-x-24 gap-y-16 mt-4">
      {contents &&
        contents.map((content) => {
          const { title, links } = content;
          return (
            <div key={title} className="dashboard-tile-container">
              <h2 className="flex items-center m-0">
                {iconsMap[title]?.()}
                <div className="ml-2">{content.title}</div>
              </h2>

              <p className="mt-6 mb-0">{content.content}</p>
              {typeof links === "object" &&
                links.map((link, i) => {
                  const key = link.href;
                  let isNotification = false;
                  const notificationType = notificationMap.hasOwnProperty(key);
                  if (notificationType) {
                    link.title =
                      notificationMap[key as keyof typeof notificationMap]?.(
                        operatorStatus
                      );
                    if (key == "/dashboard/select-operator") {
                      isNotification = !link.title.includes("0");
                    }
                  }

                  return (
                    <a
                      key={i}
                      href={link.href}
                      className={`flex items-center mt-6 no-underline ${
                        isNotification && "font-bold"
                      }`}
                    >
                      {notificationType ? (
                        <>
                          {isNotification && (
                            <span className="mr-2">
                              <Notification />
                            </span>
                          )}
                          {link.title}
                        </>
                      ) : (
                        link.title
                      )}
                    </a>
                  );
                })}
            </div>
          );
        })}
    </section>
  );
}

```

Reply:
**Note:** the current svg icons have errors in the console

For your consideration, we might use `@iconify/react` for the icons eliminating the need for the hard-coded `const iconsMap`

Something like
```
type ContentItem = {
  title: string;
  icon?: string;
  content: string;
  links?: { title: string; href: string;  }[];
};

client/app/data/dashboard/bceid/select_operator.json:
[
  {  "icon":"lucide:mail",
    "title": "Select Operator",
    "content": "View or update your operator information",
    "links": [
      {
        "title": "will be changed by function",
        "href": "/dashboard/select-operator",
      }
    ]
  }
]
```

Reply:
> **Note:** the current svg icons have errors in the console
> 
> For your consideration, we might use `@iconify/react` for the icons eliminating the need for the hard-coded `const iconsMap`
> 
> Something like
> 
> ```
> type ContentItem = {
>   title: string;
>   icon?: string;
>   content: string;
>   links?: { title: string; href: string;  }[];
> };
> 
> client/app/data/dashboard/bceid/select_operator.json:
> [
>   {  "icon":"lucide:mail",
>     "title": "Select Operator",
>     "content": "View or update your operator information",
>     "links": [
>       {
>         "title": "will be changed by function",
>         "href": "/dashboard/select-operator",
>       }
>     ]
>   }
> ]
> ```

Thanks, I have fixed the errors in the SVGs by camel casing some of the keys to make react happy. Also added the icon key to the json as suggested and simplified the icon map. Now it would be simple to use an icon library like you suggested though I'm not sure which theme these icons are from? Not sure if it's worth adding a library now that I've cleaned up the SVGs though I'm not opposed to it, we will have to ask the designers where they pulled these from.

Reply:
> if (key == "/dashboard/select-operator") {
>                       isNotification = !link.title.includes("0");
>                     }

This solution looks good except I'm not sure about hardcoding the url `if (key == "/dashboard/select-operator")` here. It could cause issues if the url gets changed in the future. Maybe we can meet tomorrow and discuss the best way to do this notification system and make it simple to extend :)